File: app/src/main/java/com/evanzheng/bibliobarcode/AuthorDeleteListener.java
package com.evanzheng.bibliobarcode;

//A custom implementation of an OnClickListener with an id field.

import android.view.View;

abstract class AuthorDeleteListener implements View.OnClickListener {
    int id;

    AuthorDeleteListener(int id) {
        this.id = id;
    }

}

File: app/src/main/java/com/evanzheng/bibliobarcode/AuthorDeleteListener.kt
package com.evanzheng.bibliobarcode;

//A custom implementation of an OnClickListener with an id field.

import android.view.View;

abstract class AuthorDeleteListener implements View.OnClickListener {
    final int id;

    AuthorDeleteListener(int id) {
        this.id = id;
    }

}
